edit-post/hooks/autocompleters/index.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * WordPress dependencies.
+ */
+import { addFilter } from '@wordpress/hooks';
+import { select } from '@wordpress/data';
+import { createBlock } from '@wordpress/blocks';
+
+// Export for unit test
+export function getBlockOptions( query ) {
+	const editor = select( 'core/editor' );
+	const isInitialQuery = ! query;
+
+	// TODO: Fetch reusable blocks if necessary (they aren't fetched initially)
+	return isInitialQuery ?
+		// Before we have a query, offer frecent blocks as a sensible default.
+		editor.getFrecentInserterItems() :
+		editor.getInserterItems();
+}
+
+// Export for unit test
+export function getBlockCompletion( inserterItem ) {
+	const { name, initialAttributes } = inserterItem;
+	return {
+		action: 'replace',
+		value: createBlock( name, initialAttributes ),
+	};
+}
+
+// Export for unit test
+export function addReusableBlocksCompletion( completers ) {
+	const blocksCompleter = completers.find( c => 'blocks' === c.name );
+	if ( blocksCompleter ) {
+		blocksCompleter.options = getBlockOptions;
+		blocksCompleter.getOptionCompletion = getBlockCompletion;
+	}
+
+	return completers;
+}
+
+addFilter(
+	'blocks.Autocomplete.completers',
+	'core/edit-post/blocks/media-upload/replaceMediaUpload',
```

Comment:
The name will probably need changing, as this name doesn't seem to match the context of this filter.

Reply:
That is embarrassing... I was trying to get something out there quickly and copied something nearby.

Reply:
... coding while watching my kids. Thanks for the heads up.